### PR TITLE
useStore hook returned on Store ref

### PR DESCRIPTION
**Pull request by rkyle35242**
Including a manner by which to consume the resulting Store on the ref allows easy mocking in tests.

Usage:

image

Tests:

|Before|After|
|---|---|
|image|image|

### CODE REVIEW

File: osmosis/src/setupStore.js [hunk 1]
```diff
@@ -107,7 +114,10 @@ const setupStore = (useCustomHook, config = {}) => {
     return Wrapper;
   };
 
+  const useStore = () => useContext(StoreContext);
+
   if (!!Proxy && config.proxyEnabled) {
+    storeProxyObject['useStore'] = useStore;
```
Comment (rkyle35242) [inline osmosis/src/setupStore.js:120]: Returning this from the Proxy in a similar fashion to `Context` and `Provider` makes this difficult/impossible to mock due to the nature of Proxies. Setting it on the ref makes this simple.

File: osmosis/src/setupStore.js [hunk 2]
```diff
@@ -48,6 +55,7 @@ const setupStore = (useCustomHook, config = {}) => {
     let store = useCustomHook(props);
     if (!!store.Context) throw new Error("'Context' property is protected and cannot exist on a store object");
     if (!!store.Provider) throw new Error("'Provider' property is protected and cannot exist on a store object");
+    if (!!store.useStore) throw new Error("'Provider' property is protected and cannot exist on a store object");
```
Comment (chaceburnette) [inline osmosis/src/setupStore.js:58]: error message here needs updating

Comment (rkyle35242) [reply osmosis/src/setupStore.js:58]: Ha, thank you

**Comment by rkyle35242** [comment]
Confirmed after changing to the get trap override that it works:

image

**Pull request merged**